### PR TITLE
added documentation for docfx generated developer documentation

### PR DESCRIPTION
**Pull request by thalbern**
Overview
--- added documentation for docfx generated developer documentation
--- added entry in TOC file

### CODE REVIEW

File: Documentation/DevDocGuide.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+* open a webbrowser with localhost:*portnumber*
+
+Note that on executing the docfx command on the json build file docfx will show any broken links in the documentation as warning. 
+Please make sure whenever you perform changes on any of the documentation files or API to update all links pointing to these articles or code.
+
+## Using crefs and hrefs in /// documented code
+Docfx supports crefs in /// documented code. It will translate those references to links pointing to the generated api documentation or to external documentation websites.
+External xref services for resolving links to external libraries/apis can be added to the docfx.json build settings file in the property *xrefService*.
+
+For external apis that don't provide an xref service hrefs to the documentaion website can be added to the comments.
```
Comment (wiwei) [inline Documentation/DevDocGuide.md]:
```suggestion
For external apis that don't provide an xref service hrefs to the documentation website can be added to the comments.
```

**Comment by chrisfromwork** [comment]
It may be worth adding some stub calling out our documentation to the repo landing page as well:
https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/mrtk_release/README.md

File: Documentation/DevDocGuide.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+### Setup
+* get the latest version of [docfx](https://dotnet.github.io/docfx/index.html)
+* extract the files in a folder on your computer
+* add the folder to your PATH in your environment variables
+
+### Generation
+* open a powershell or cmd prompt in the root of the MRTK project
+* execute docfx docfx.json (optionally with the -f option to force a rebuild of doc files)
+* execute docfx serve doc (optionally with -p *portnumber* if you don't want to use the 8888 default port)
+* open a webbrowser with localhost:*portnumber*
```
Comment (MenelvagorMilsom) [inline Documentation/DevDocGuide.md]:
```suggestion
* open a web browser with localhost:*portnumber*
```

**Comment by thalbern** [comment]
> It may be worth adding some stub calling out our documentation to the repo landing page as well:
> https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/mrtk_release/README.md

I've added it to the README in mrtk_development branch

**Pull request merged**